### PR TITLE
DirListing: Refactor selection so that it's based on path not name

### PR DESCRIPTION
**Pull request by martinRenou**
The fact that the selection is based on the item name is problematic for custom filebrowsers (e.g. filetree browsers), because the name might not be unique across directories

### CODE REVIEW

File: packages/filebrowser/src/listing.ts [hunk 1]
```diff
@@ -1646,8 +1628,8 @@ export class DirListing extends Widget {
   private _clipboard: string[] = [];
   private _manager: IDocumentManager;
   private _softSelection = '';
-  private _selection: { [key: string]: boolean } = Object.create(null);
-  private _renderer: DirListing.IRenderer;
+  protected _selection: { [key: string]: boolean } = Object.create(null);
+  protected _renderer: DirListing.IRenderer;
```
Comment (martinRenou) [inline packages/filebrowser/src/listing.ts:1632]: Also making these two protected so that they can be accessed in subclasses

**Review by blink1073 — APPROVED** [review]
Thanks!

**Comment by blink1073** [comment]
Kicking CI

File: packages/filebrowser/src/listing.ts (hunk 1 above)
Comment (krassowski) [reply packages/filebrowser/src/listing.ts:1648]: Would it make sense to completely expose the renderer (public) as for: https://github.com/jupyterlab/jupyterlab-git/pull/885 ?

Comment (martinRenou) [reply packages/filebrowser/src/listing.ts:1648]: I guess so.

I start to believe this renderer should not be mandatory for DirListing subclasses. The current IRenderer is really bound to the way things are done in the core filebrowser currently (a list of files in one directory), which makes it very difficult to subclass and extend for e.g. a file tree logic.

**Pull request merged**